### PR TITLE
Multiple fixes for various moderation features

### PR DESCRIPTION
**Pull request by applenick**
# May Moderation Fixes 🎉 
This PR contains fixes for several moderation bugs as well as a few small new features.

Overview of changes:
* Frozen players who are kicked, are now automatically unfrozen.
* Add a `-s` flag to `/freeze` command, allowing the freezer to hide their identity. (Vanished will automatically be silent too)
* Rewrote logic for `/reports`, which fixes relogged users having an unformatted name.
* Muted players can now `/msg` staff (fixing a reported bug where sounds play with no message)
* Changed `/setnext` map name color to `GOLD`
* `/setnext -r` will now unset any map that was setnext
* Fix vanished players showing up in `/staff` & `/match`
* Fix vanished players joining before match start and not being forced onto obs
* Fix kick/ban screen missing config
* Fix missing translations (there were a few)
* Moved `PlayerReportEvent` to community package
* Migrate 95% of community module to the new text library
* Vanished players will no longer have their chat setting forcefully set, instead global and team chat will be overridden if a vanished player attempts to use them, and redirected into admin chat.
* Switch /vanish and /freeze to toggle their respective features (removes /unvanish and /unfreeze)

Signed-off-by: applenick <applenick@users.noreply.github.com>

### CODE REVIEW

File: core/src/main/java/tc/oc/pgm/community/commands/ModerationCommands.java [hunk 1]
```diff
@@ -98,124 +98,116 @@ private void removeCachedBan(BannedAccountInfo info) {
...
 
     Component content =
         onlineStaff.isEmpty()
-            ? new PersonalizedTranslatable("moderation.staff.empty")
-                .getPersonalizedText()
-                .color(ChatColor.RED)
-            : new Component(
-                Components.join(new PersonalizedText(", ").color(ChatColor.GRAY), onlineStaff));
+            ? TranslatableComponent.of("moderation.staff.empty")
+            : TextComponent.join(TextComponent.of(", ", TextColor.GRAY), onlineStaff);
```
Comment (Electroid) [inline core/src/main/java/tc/oc/pgm/community/commands/ModerationCommands.java:120]: `TextFormatter.list`

File: core/src/main/java/tc/oc/pgm/community/commands/ModerationCommands.java [hunk 2]
```diff
@@ -546,30 +532,27 @@ public void alts(
...
-              headerText,
-              new PersonalizedText(" (").color(ChatColor.GRAY),
-              new PersonalizedText(Integer.toString(altAccounts.size())).color(ChatColor.DARK_AQUA),
-              new PersonalizedText(") » Page ").color(ChatColor.GRAY),
-              pageNum);
+          TextComponent.builder()
+              .append(headerText)
+              .append(" (", TextColor.GRAY)
+              .append(Integer.toString(altAccounts.size()), TextColor.DARK_AQUA)
+              .append(") » Page ", TextColor.GRAY)
```
Comment (Electroid) [inline core/src/main/java/tc/oc/pgm/community/commands/ModerationCommands.java:548]: Can we use the page translations we have? It's either under `command` or `misc`.

File: core/src/main/java/tc/oc/pgm/community/commands/ModerationCommands.java [hunk 3]
```diff
@@ -824,71 +780,74 @@ public static String formatPunishmentScreen(
...
+        TextComponent.builder().append(WARN_SYMBOL).append(titleWord).append(WARN_SYMBOL).build();
+    Component subtitle = formatPunishmentReason(reason).color(TextColor.GOLD);
+
+    // TODO: Upgrade show title to new text
+    tc.oc.pgm.util.component.Component oldTitle =
+        new PersonalizedText(TextTranslations.translateLegacy(title, target.getBukkit()));
+    tc.oc.pgm.util.component.Component oldSubTitle =
+        new PersonalizedText(TextTranslations.translateLegacy(subtitle, target.getBukkit()));
+
+    target.showTitle(oldTitle, oldSubTitle, 5, 200, 10);
```
Comment (Electroid) [inline core/src/main/java/tc/oc/pgm/community/commands/ModerationCommands.java:850]: I would modify `Audience` to add `showTitle` and the hotbar with the new components. Make them defaults and use `TextTranslations`

File: core/src/main/java/tc/oc/pgm/community/commands/ModerationCommands.java [hunk 4]
```diff
@@ -980,12 +936,14 @@ public Component getPunisher() {
...
-          PeriodFormats.relativePastApproximate(java.time.Instant.ofEpochMilli(time.toEpochMilli()))
-              .color(ChatColor.DARK_AQUA);
-
-      return new PersonalizedTranslatable("moderation.similarIP.hover", getPunisher(), timeAgo)
-          .getPersonalizedText()
-          .color(ChatColor.GRAY);
+          TextComponent.of(
+              ComponentRenderers.toLegacyText(
+                  PeriodFormats.relativePastApproximate(
+                          java.time.Instant.ofEpochMilli(time.toEpochMilli()))
```
Comment (Electroid) [inline core/src/main/java/tc/oc/pgm/community/commands/ModerationCommands.java:942]: Just use `time` here?

File: core/src/main/java/tc/oc/pgm/community/commands/ReportCommands.java [hunk 5]
```diff
@@ -149,98 +144,115 @@ public void reportHistory(
...
+      return getUsername(getTarget(), match);
+    }
+
+    public Component getSenderComponent(Match match) {
+      return getUsername(getSender(), match);
+    }
+
+    private Component getUsername(UUID uuid, Match match) {
+      MatchPlayer player = match.getPlayer(targetUUID);
+      Component name = TextComponent.of("Unknown", TextColor.AQUA, TextDecoration.ITALIC);
```
Comment (Electroid) [inline core/src/main/java/tc/oc/pgm/community/commands/ReportCommands.java:249]: `misc.unknown` I believe exists

File: util/src/main/i18n/templates/command.properties [hunk 6]
```diff
@@ -71,4 +71,9 @@ command.list.online = Total Online: {0}
 
 command.list.teams = Teams:
 
-command.list.participants = Participants
\ No newline at end of file
+command.list.participants = Participants
+
+command.message.noReply = Did not find a message to reply to, use /msg
```
Comment (Electroid) [inline util/src/main/i18n/templates/command.properties:76]: `/msg` should be a `{0}`

File: util/src/main/i18n/templates/map.properties [hunk 7]
```diff
@@ -50,6 +50,12 @@ map.setNext = Next map set to {0} by {1}
 
 map.setNext.confirm = You may not set the next map when a restart is queued. Use -f to override.
 
+# {0} = player name (who unset the map)
+# {1} = Name of map that was unset.
```
Comment (Electroid) [inline util/src/main/i18n/templates/map.properties:54]: lowercase, no periods

File: util/src/main/i18n/templates/map.properties [hunk 8]
```diff
@@ -50,6 +50,12 @@ map.setNext = Next map set to {0} by {1}
...
+map.setNext.revert = {0} has removed {1} as the next map.
+
+map.setNext.none = No map has been set next.
```
Comment (Electroid) [inline util/src/main/i18n/templates/map.properties:57]: Same as `map.noNextMap`?

File: util/src/main/i18n/templates/map.properties [hunk 9]
```diff
@@ -89,10 +95,15 @@ pool.title = Map Pools
 
 pool.noPoolMatch = No map pools matched query
 
+pool.noDynamic = No dynamic pool could be found.
```
Comment (Electroid) [inline util/src/main/i18n/templates/map.properties:98]: Generally, no periods in translation keys

**Comment by applenick** [comment]
Just pushed some more changes. I adjusted how the join/quit messages work for vanished players, as I feel the changes between my original PR and the merge made things a bit confusing. 

Now when a player vanishes/appears, regular players will see a message and staff will not whereas currently staff receive the `quietly joined/quit` upon every vanish action. In addition to that when a vanished player actually quits/rejoins, staff members will receive a broadcast via admin chat stating the player has quietly joined/quit. I feel that it is more important for staff to view when other staff actually join/leave over when they vanish. 🤷‍♂️ 

I’m still working on going through as many things as I can to test. But my guess is later today everything should be good to go, pending any other changes.

**Pull request merged**